### ISSUE TITLE
Void elements are given closing tags

### ISSUE THREAD

**Issue by mjbshaw** [htacg/tidy-html5#162]
Given the following input file:

```html
<html>
<head>
<title>Hello</title>
</head>
<body>
<p>
Hello world!
<p>
How are you?
<p> Good!
<img alt="baz" src="baz.png"/>
<img alt="foo" src="foo.png">
<br/>
<br>
foo:bar>asdf/foo:bar>
</body></html>
```

Running `./tidy5 -access 1 --vertical-space f -wrap 0 file.html` produces:

```html
<!DOCTYPE html>
<html>
<head>
<title>Hello</title>
</head>
<body>
<p>Hello world!</p>
<p>How are you?</p>
<p>Good! <img alt="baz" src="baz.png"></img> <img alt="foo" src="foo.png"><br></br>
<br>
foo:bar&gt;asdf/foo:bar&gt;</p>
</body>
</html>
```

Tidy shouldn't add closing tags for void elements (note the `</img>` and `</br>`), as those aren't legal in HTML5.

**Comment by geoffmcl** [comment]
@mjbshaw, @ermshiperete, have pushed a fix for this.

This occurred during the fix for Issue #111 (#112 & #113), adding an end tag to ALL elements marked StartEnd when outputting HTML5. Have left that in place but excepted all the void elements listed here http://www.w3.org/TR/html-markup/syntax.html#syntax-elements...

Established a new service isVoidElement(node) which returns yes if in the list... maybe this service could be optimized more...

Would be pleased if you could both get a chance to retest, and if ok, close this issue.

Bumped version to 4.9.17 in the develop-500 branch for this fix.

**Comment by mjbshaw** [comment]
Looks good. Output is now:

```html
<!DOCTYPE html>
<html>
<head>
<title>Hello</title>
</head>
<body>
<p>Hello world!</p>
<p>How are you?</p>
<p>Good! <img alt="baz" src="baz.png"> <img alt="foo" src="foo.png"><br>
<br>
foo:bar&gt;asdf/foo:bar&gt;</p>
</body>
</html>
```
